### PR TITLE
Ability to filter clusters by using allow list and block list

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Ability to filter clusters by using allow list and block list. All cases are covered by unit tests.

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Unit tests (no changes in the code)

## Testing steps

Done on CI.

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/cluster_filter.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			// -> skip rest of the loop
+			if collections.StringInSlice(clusterName, configuration.BlockedClusters) {
+				stat.Blocked++
+				// don't do anything else with the cluster
+				continue
+			}
+			// not blocked
+			// -> put it into the output list
+			stat.Filtered++
+			filtered = append(filtered, cluster)
```
Comment (dpensi) [inline differ/cluster_filter.go:103]: I'm not sure we're supposed to add clusters here, what if they are not in the allow list or what if they have been already added in the previous if (line 76)

Comment (tisnik) [reply differ/cluster_filter.go:103]: If they are not in allow list, it wont be added because of `continue` on line 86

**Review by joselsegura — APPROVED** [review]
The behaviour when both allow and block lists are enabled can be unexpected (having a cluster in both lists will "allow" it, having a cluster that is not in any of the lists, will allow it too), but the logic seems fine to me

**Comment by tisnik** [comment]
> The behaviour when both allow and block lists are enabled can be unexpected (having a cluster in both lists will "allow" it, having a cluster that is not in any of the lists, will allow it too), but the logic seems fine to me

If the cluster is not in allow list, it is not added. Basically turning on allow list blocks all other clusters.

**Review by dpensi — APPROVED** [review]

**Comment by joselsegura** [comment]
Oh, I see now. At the end, if the allow list is enabled, the block list is ignored, isn't it?

File: differ/cluster_filter_test.go [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+	// check filter output
+	assert.Empty(t, filtered)
+	assert.Equal(t, 0, stat.Input)
+	assert.Equal(t, 0, stat.Allowed)
+	assert.Equal(t, 0, stat.Blocked)
+	assert.Equal(t, 0, stat.Filtered)
+}
+
+// TestFilterNoFilters test checks filtering if filters are disabled.
+// In this test case, list with three clusters is provided at input.
```
Comment (epapbak) [inline differ/cluster_filter_test.go:106]: four clusters

**Pull request merged**